### DISCUSSION TITLE
handling query params for getStaticPaths

### DISCUSSION THREAD

**Discussion by csenio** [vercel/next.js#11487] [Help]
#### The issue:
`getStaticPaths` right now only works for actual routes like `/blog/page/[page]`. I'd like to use it with query params like `/blog?page=1`. 

#### Solutions I considered:
 - making the entire thing the param (`/[blogWithQuery] -> /blog?page=1`), but that would prevent me from putting another dynamic page at the root (like `/[gender]/products`)
 - putting the query parameter into the route param (`/blog/[query] -> /blog/?page=1`) but that's a bit unusual and I had clients complain about it.
 - fetching pages dynamically or at run time, but that leads to slowdowns or worse ux.

Are there any recommended, or alternative methods for handling this?

**Comment by timneutkens** [comment]
getStaticProps / getStaticPaths generate static html files. Static html files generally can't access the querystring without significant server/proxy instrumentation which would lock in a solution.

**Reply by petewarman** [reply]
I have an app with a page `/furniture` which displays a list of items and `/furniture/[id]` which displays individual item details.

The items displayed in `/furniture` are grouped into one of 6 categories. I'd like to be able to filter the items on the page by category like so `/furniture?category=dining-table`. 

I understand where you have a user defined search term or something that using query params in getStaticPaths & getStaticProps makes no sense, but for my use case when there is a known list of possible values it would be handy to be able to use query params in these methods.

**Comment by lucas-soaresc** [comment]
any answers about that?
